File: cmd/kill.go
package cmd

import (
	"context"
	"fmt"

	"portkill/internal/finder"
	"portkill/internal/formatter"
	"portkill/internal/killer"
	"portkill/internal/port"

	"github.com/spf13/cobra"
)

func init() {
	rootCmd.AddCommand(killCmd)
}

var killCmd = &cobra.Command{
	Use:   "kill <port>",
	Short: "Kill process running on a specific port",
	Args:  cobra.ExactArgs(1),
	RunE: func(cmd *cobra.Command, args []string) error {
		p, err := port.ParsePort(args[0])
		if err != nil {
			return fmt.Errorf("invalid port: %w", err)
		}

		ctx := context.Background()
		processes, err := finder.FindPort(ctx, p)
		if err != nil {
			return fmt.Errorf("failed to find processes: %w", err)
		}

		if len(processes) == 0 {
			fmt.Printf("No process found on port %d\n", p)
			return nil
		}

		formatter.PrintProcesses(processes, flagVerbose)

		if !flagYes && !formatter.Confirm(processes) {
			fmt.Println("Aborted")
			return nil
		}

		k := killer.NewKiller()
		var lastErr error
		for _, p := range processes {
			if err := k.Kill(ctx, p.PID, flagForce); err != nil {
				fmt.Fprintf(os.Stderr, "Failed to kill PID %d: %v\n", p.PID, err)
				lastErr = err
			} else {
				fmt.Printf("Killed PID %d (%s) on port %d\n", p.PID, p.Name, p.Port)
			}
		}

		return lastErr
	},
}

File: src/commands/kill.js
const { Command } = require('commander');
const { createFinder, createKiller } = require('../index');
const { printProcesses, confirm, parsePort, styles } = require('../utils/formatter');

const killCmd = new Command('kill');
killCmd
  .description('Kill process running on a specific port')
  .argument('<port>', 'Port number')
  .action(async (portStr) => {
    const port = parsePort(portStr);
    const options = killCmd.parent.opts();
    const force = options.force || false;
    const verbose = options.verbose || false;
    const yes = options.yes || false;

    const finder = createFinder();
    const killer = createKiller();

    let processes;
    try {
      processes = await finder.findByPort(port);
    } catch (err) {
      console.error(`${styles.error('Error:')} Failed to find processes: ${err.message}`);
      process.exit(1);
    }

    if (processes.length === 0) {
      console.log(styles.muted(`No process found on port ${port}`));
      return;
    }

    printProcesses(processes, verbose);

    if (!yes) {
      const confirmed = await confirm(processes);
      if (!confirmed) {
        console.log(styles.muted('Aborted'));
        return;
      }
    }

    let hasError = false;
    for (const p of processes) {
      try {
        await killer.kill(p.pid, force);
        console.log(styles.success(`✓ Killed ${p.name} (PID: ${p.pid}) on port ${p.port}`));
      } catch (err) {
        console.error(styles.error(`✗ Failed to kill PID ${p.pid}: ${err.message}`));
        hasError = true;
      }
    }

    if (hasError) {
      process.exit(1);
    }
  });

module.exports = killCmd;
